src/Context/MessageContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useReducer, useContext, createContext, Dispatch } from 'react';
+
+type Message = {
+        userID: number;
+        text: string;
+    };
+
+type Action =
+    | { type: 'ADD' };
+
+type MessageDispatch = Dispatch<Action>;
+
+const MessageStateContext = createContext<Message[]>([{
+    userID : 0,
+    text: 'Dummy'
+}]);
+const MessageDispatchContext = createContext<MessageDispatch | null >(null);
+
+const initialMessages = [
+    {
+        userID : 0,
+        text: 'Sly bunny'
+    },
+    {
+        userID : 1,
+        text: 'Dumb fox.'
+    },
+    {
+        userID : 0,
+        text: 'Come on, you know I love you'
+    }
+];
+
+function messageReducer(state: Message[], action: Action) {
+    switch (action.type) {
+        case "ADD":
+            return state.concat({ 
+                    userID : 1, 
+                    text: 'Okay' });
```

Comment:
`spread operator`를 사용해 봅시다

```suggestion
           return [
        ...state,
        {
          userID: 1,
          text: 'Okay',
        },
      ];
```